torch/csrc/jit/passes/quantization.cpp

```diff
@@ -572,5 +599,107 @@ template TORCH_API void InsertQuantDequantNodesForParam(
         getQParamFunc,
     at::ScalarType t);
 
+
+static Node* prepQuantAddObserverFor(
+    Value* v,
+    Node* original_observer_node,
+    Node* insert_point) {
+  TORCH_INTERNAL_ASSERT(insert_point != nullptr);
+  WithInsertPoint ins(insert_point);
+
+  // Create a new observer node. We just need to clone the original one.
+  Node* observerNode = insert_point->owningGraph()->createClone(
+      &*original_observer_node, [&](Value* v) { return v; }, false);
+
+  // Set the type and the name of the output of the new observer node. It will
+  // be used instead of the original value v.
+  Value* observedValue = observerNode->addOutput();
+  observedValue->setType(v->type());
+  observedValue->setDebugName(v->debugName() + ".observed");
+
+  // Now we can add the inputs.
+  observerNode->addInput(v);
+  return observerNode;
+}
+
+TORCH_API script::Module PrepareQuant(
```

Comment:
I think the name of this function should be more specific - "prepare" can mean anything. I realize that it comes from how we describe it to users as they go through quantization process, but I think here in the code it doesn't have to use the same one.

Reply:
ok, I'll change to insertobserver